content/articles/react-drf-shortener/index.md

```diff
@@ -0,0 +1,361 @@
+            method: "POST",
+            body: JSON.stringify({ longurl: longurl }),
+            headers: { "Content-Type": "application/json" },
+        })
+            .then((res) => res.json())
+            .then((data) => {
+                setShorturl(data.shorturl);
+                setReturnLongURL(data.longurl);
+                setLongurl("");
+            });
+    };
+
+    return (
+        <div style={{ textAlign: "center" }}>
+            <input
+                type="text"
+                name="longurl"
+                value={longurl}
+                onChange={(e) => setLongurl(e.target.value)}
+            />
+            <button
+                type="submit"
+                onClick={(e) => handleSubmit(e)}
+                disabled={!longurl}
+            >
+                shorten
+            </button>
+
+            <div>
+                <p>Long URL: {returnLongURL}</p>
+                <p
+                    style={{ cursor: "pointer" }}
+                    onClick={() => window.open(returnLongURL)}
+                >
+                    Short URL: {shorturl}
+                </p>
+            </div>
+        </div>
+    );
+}
+
+export default App;
+```
+
+Code explanation:
+
+We import the `useState()` hook from `react` in the code above. We then set up three pieces of state as explained below:
+
+-   `longurl` - this will be used to track the input provided by the user.
+-   `shorturl` - this will be used to hold the value of the short URL returned by the server.
+-   `returnLongURL` - this will be used to hold the value of the long URL provided by the server.
+
+In the `return` section, there is an input field and a button that calls the `handleSubmit()` function when clicked. This function uses the `fetch API` to send the `longurl` to the backend. The `fetch API` returns a response and sets the `shorturl` and `returnLongURL` values.
+
+The long URL and the short URL are rendered on the browser. We use the `window.open()` method to open the long URL on a new tab at the click of the short URL.
+
+To run the front-end code, type in the following command in the integrated terminal and press enter:
+
+```bash
+npm start
+```
+
+The above command will open a local development server on `127.0.0.1:3000`
+
+#### Testing the Shortener Service.
+
+To test the service, open the browser and navigate to the following URL, `127.0.0.1:3000`. Make sure that server is running. Input the long URL on the input field provided and click on the `shorten` button. A short URL will be rendered, and after clicking on it, it redirects to the same webpage as the long URL. This means that our shortener service is working.
+
+#### Conclusion
+
+We have how to implement a simple shortener service using React and Django. However, this project can be taken to the next level by styling the user interface and adding more features. One of the features that may be added is counting the number of times a link has been created.
```

Comment:
Can you add a GitHub URL containing source code? If possible, you can trying hosting this with GitHub & Netlify.